test/linter.test.js

```diff
@@ -38,30 +47,29 @@ describe('linter.js', () => {
     describe('lint()', () => {
         const profilesDir = path.join(__dirname, './profiles/');
 
-        fs.readdirSync(profilesDir).forEach(function (file) {
-            const profile = JSON.parse(fs.readFileSync(profilesDir + file, 'utf8'))
-            const profileName = file.replace(path.extname(file), '')
+        ['default', 'strict'].forEach(profileName => {
+            const profile = JSON.parse(fs.readFileSync(profilesDir + profileName + '.json', 'utf8'))
 
             context('when `' + profileName + '` profile is loaded', () => {
-                testProfile(profile);
+                profile.fixtures.forEach(fixture => {
+                    describe(`linting the ${fixture.object} object`, () => {
+                        testFixture(fixture, profile.rules);
+                    });
+                });
             });
         })
 
         context('when rules are manually passed', () => {
-
             const lintAndExpectErrors = (rule, input, expectedErrors) => {
-                const options = { lintResults : []};
-                linter.setRules([rule]);
-                linter.lint('something', input, options);
-                const ruleErrors = options.lintResults.map(result => result.rule.name);
-                ruleErrors.should.deepEqual(expectedErrors);
+                linter.initialize();
```

Comment:
New API for linter is easier to work with thanks to awesome advice from @MikeRalphson.